payjoin/src/uri.rs

```diff
@@ -53,6 +53,44 @@ pub trait UriExt<'a>: sealed::UriExt {
     fn check_pj_supported(self) -> Result<PjUri<'a>, bip21::Uri<'a>>;
 }
 
+impl<'a> bip21::SerializeParams for &'a Payjoin {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        match self {
+            Payjoin::Supported(params) => params.serialize_params(),
+            Payjoin::V2Only(params) => params.serialize_params(),
+            Payjoin::Unsupported => vec![].into_iter(),
+        }
+    }
+}
+
+impl<'a> bip21::SerializeParams for &'a PayjoinParams {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        vec![
+            ("pj", self._endpoint.as_str().to_string()),
+            ("pjos", if self.disable_output_substitution { "1" } else { "0" }.to_string()),
+            #[cfg(feature = "v2")]
+            (
+                "ohttp",
+                if self.ohttp_config.is_some() {
+                    PjUriBuilder::encode_ohttp_config(self.ohttp_config.as_ref().unwrap())
+                        .unwrap_or_default()
+                } else {
+                    "".to_string()
```

Comment:
not really sure about this..

Reply:
If `self.ohttp_config.is_none()` we shouldn't add an element to the iterator at all